Makefile

```diff
@@ -1,428 +0,0 @@
-SAIL_FLAGS += -dno_cast
-SAIL_DOC_FLAGS ?= -doc_embed plain
-
-# Sail command to use.
-SAIL := sail
-
-# <sail install dir>/share/sail
-SAIL_DIR := $(shell $(SAIL) --dir)
-SAIL_LIB_DIR := $(SAIL_DIR)/lib
-SAIL_SRC_DIR := $(SAIL_DIR)/src
-
-LEM_DIR := $(SAIL_DIR)/../lem
-export LEM_DIR
-
-C_WARNINGS ?=
-#-Wall -Wextra -Wno-unused-label -Wno-unused-parameter -Wno-unused-but-set-variable -Wno-unused-function
-C_INCS = $(addprefix c_emulator/,riscv_prelude.h riscv_platform_impl.h riscv_platform.h riscv_softfloat.h)
-C_SRCS = $(addprefix c_emulator/,riscv_prelude.c riscv_platform_impl.c riscv_platform.c riscv_softfloat.c riscv_sim.c)
-
-SOFTFLOAT_DIR    = dependencies/softfloat/berkeley-softfloat-3
-SOFTFLOAT_INCDIR = $(SOFTFLOAT_DIR)/source/include
-SOFTFLOAT_LIBDIR = $(SOFTFLOAT_DIR)/build/Linux-RISCV-GCC
-SOFTFLOAT_FLAGS  = -I $(SOFTFLOAT_INCDIR)
-SOFTFLOAT_LIBS   = $(SOFTFLOAT_LIBDIR)/softfloat.a
-SOFTFLOAT_SPECIALIZE_TYPE = RISCV
-
-GMP_FLAGS = $(shell pkg-config --cflags gmp)
-# N.B. GMP does not have pkg-config metadata on Ubuntu 18.04 so default to -lgmp
-GMP_LIBS = $(shell pkg-config --libs gmp || echo -lgmp)
-
-# TODO: Remove Zlib when upgrading to Sail 0.19; it is no longer a requirement.
-ZLIB_FLAGS = $(shell pkg-config --cflags zlib)
-ZLIB_LIBS = $(shell pkg-config --libs zlib)
-
-C_FLAGS = -I $(SAIL_LIB_DIR) -I c_emulator $(GMP_FLAGS) $(ZLIB_FLAGS) $(SOFTFLOAT_FLAGS)
-C_LIBS  = $(GMP_LIBS) $(ZLIB_LIBS) $(SOFTFLOAT_LIBS)
-
-# The C simulator can be built to be linked against Spike for tandem-verification.
-# This needs the C bindings to Spike from https://github.com/SRI-CSL/l3riscv
-# TV_SPIKE_DIR in the environment should point to the top-level dir of the L3
-# RISC-V, containing the built C bindings to Spike.
-# RISCV should be defined if TV_SPIKE_DIR is.
-ifneq (,$(TV_SPIKE_DIR))
-C_FLAGS += -I $(TV_SPIKE_DIR)/src/cpp -DENABLE_SPIKE
-C_LIBS  += -L $(TV_SPIKE_DIR) -ltv_spike -Wl,-rpath=$(TV_SPIKE_DIR)
-C_LIBS  += -L $(RISCV)/lib -lfesvr -lriscv -Wl,-rpath=$(RISCV)/lib
-endif
-
-# SAIL_FLAGS = -dtc_verbose 4
-
-ifneq (,$(COVERAGE))
-C_FLAGS += --coverage -O1
-SAIL_FLAGS += -Oconstant_fold
-else
-C_FLAGS += -O3 -flto=auto
-endif
-
-ifneq (,$(SAILCOV))
-ALL_BRANCHES = generated_definitions/c/all_branches
-C_FLAGS += -DSAILCOV
-SAIL_FLAGS += -c_coverage $(ALL_BRANCHES) -c_include sail_coverage.h
-C_LIBS += $(SAIL_LIB_DIR)/coverage/libsail_coverage.a -lm -lpthread -ldl
-endif
-
-# Optionally link C_LIBS statically. Unlike -static this will not
-# link glibc statically which is generally a bad idea.
-ifneq (,$(STATIC))
-    UNAME_S := $(shell sh -c 'uname -s 2>/dev/null || echo not')
-    ifeq ($(UNAME_S),Darwin)
-        # Unfortunately the Mac linker does not support -Bstatic.
-        GMP_LIBS = $(shell pkg-config --variable=libdir gmp)/libgmp.a
-        C_LIBS_WRAPPED = $(C_LIBS)
-    else
-        C_LIBS_WRAPPED = -Wl,--push-state -Wl,-Bstatic $(C_LIBS) -Wl,--pop-state
-    endif
-else
-    C_LIBS_WRAPPED = $(C_LIBS)
-endif
```

Comment:
I believe we have lost support for these options. I've used the sailcov option before, but not the other ones.